edx_proctoring/api.py

```diff
@@ -1623,7 +1613,12 @@ def _get_timed_exam_view(exam, context, exam_id, user_id, course_id):
         # check if the exam's due_date has passed. If so, return None
         # so that the user can see their exam answers in read only mode.
         if not exam['hide_after_due'] and is_exam_passed_due(exam, user=user_id):
-            return None
+            if context.get('grace_period', False):
+                is_updated = verify_and_add_wait_deadline(context, exam['due_date'])
```

Comment:
what is this variable? `is_updated`?  sounds like it should be `due_date_passed`?

Reply:
nit?
```
if context.get('grace_period', False) and not exam_has_due_date(context, exam['due_date']):
    return
```

Reply:
`is_updated` signifies if the context was updated with `wait_deadline`. I think it should be renamed to `is_context_updated` to show context has been updated with something.

Reply:
and the context would only be updated there exam due date (due+grace period) correct?

Reply:
Well, yes. But naming it exam due date is not correct. This condition is for the cases where the exam submission exists. If a user tries to attempt after the due date, but with an active grace period, they will see the expired time banner.